android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/Utils.java

```diff
@@ -228,6 +228,49 @@ public static void restoreBlocked(Context context) {
         context).getStringSet("blocked", new HashSet<String>()));
   }
 
+  /**
```

Comment:
We're doing a lot of splitting and reparsing the same string...  Create some WifiDirectInfo internal class that parse method will populate with info.  it can throw an exception if it's not a valid name.